api/src/main/java/com/github/retrooper/packetevents/protocol/world/chunk/LightData.java

```diff
@@ -0,0 +1,162 @@
+package com.github.retrooper.packetevents.protocol.world.chunk;
+
+import com.github.retrooper.packetevents.manager.server.ServerVersion;
+import com.github.retrooper.packetevents.wrapper.PacketWrapper;
+
+import java.util.BitSet;
+
+public class LightData implements Cloneable {
+    private boolean trustEdges;
+    private BitSet blockLightMask;
+    private BitSet skyLightMask;
+    private BitSet emptyBlockLightMask;
+    private BitSet emptySkyLightMask;
+    private int skyLightCount;
+    private int blockLightCount;
+    private byte[][] skyLightArray;
+    private byte[][] blockLightArray;
+
+    public void read(PacketWrapper<?> packet) {
```

Comment:
in my opinion these read methods should be made static and return the `LightData` object, this allows easy usage when e.g. reading lists